onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -3257,10 +3257,65 @@ def aten_matrix_power(self: TensorType, n: int) -> TensorType:
     raise NotImplementedError()
 
 
-def aten_max(self: TensorType) -> TensorType:
+@torch_op("aten::max", trace_only=True)
+def aten_max(self: TReal, dim_or_other: TReal = None, keepdim: BOOL = None) -> TReal:
     """max(Tensor self) -> Tensor"""
 
-    raise NotImplementedError()
+    self_rank = op.Size(op.Shape(self))
+    if self_rank == 0:
+        self = op.Reshape(self, op.Constant(value_int=[-1]))
+
+    output = 1
```

Comment:
If we are really recording output number, I suggest we put output=2 here, and we can save one line in the following if/else condition ...

Reply:
refine the logic in another way